app/controllers/assets.php

```diff
@@ -39,4 +39,49 @@ function list_json() {
 
         echo json_encode($assets_list);
     }
+
+    function update() {
+        $is_success = null;
+
+        $id = null;
+        $name = null;
+        $description = null;
+
+        if (!empty($_GET['id'])) {
+            $id = $_REQUEST['id'];
+        }
+
+        if ( null == $id ) {
+            //TODO: Set warning or maybe ignore
+            header("Location: view");
+        }
+
+        if (isset($_POST['submit'])) {
+            $id = $_POST['id'];
+            $is_success = $this->assets_model->update(
+                $_POST['id'], // asset id
```

Comment:
Isn’t directly passing post parameters a problem as there is no check to check whether they are present?